xarray/core/resample.py

```diff
@@ -0,0 +1,339 @@
+    for handling specialized re-sampling operations.
+
+    You should create a `Resample` object by using the `DataArray.resample` or
+    `Dataset.resample` methods.
+
+    See Also
+    --------
+    DataArray.resample
+    Dataset.resample
+
+    """
+
+    def _upsample(self, method, *args, **kwargs):
+        """Dispatch function to call appropriate up-sampling methods on
+        data.
+
+        This method should not be called directly; instead, use one of the
+        wrapper functions supplied by `Resample`.
+
+        Parameters
+        ----------
+        method : str {'asfreq', 'pad', 'ffill', 'backfill', 'bfill', 'nearest',
+                 'interpolate'}
+            Method to use for up-sampling
+
+        See Also
+        --------
+        Resample.asfreq
+        Resample.pad
+        Resample.backfill
+        Resample.interpolate
+
+        """
+
+        _upsampled_means = self.mean(self._dim)
+
+        # Drop non-dimension coordinates along the resampled dimension
+        for k, v in self._obj.coords.items():
+            if k == self._dim:
+                continue
+            if self._dim in v.dims:
+                self._obj = self._obj.drop(k)
+
+        if method == 'asfreq':
+            return _upsampled_means
+
+        elif method in ['pad', 'ffill', 'backfill', 'bfill']:
+            kwargs = kwargs.copy()
+            kwargs.update(**{self._dim: _upsampled_means[self._dim]})
+            return self._obj.reindex(method=method, *args, **kwargs)
+
+        elif method == 'interpolate':
+            return self._interpolate(*args, **kwargs)
+
+        else:
+            raise ValueError('Specified method was "{}" but must be one of'
+                             '"asfreq", "ffill", "bfill", or "interpolate"'
+                             .format(method))
+
+    def asfreq(self):
+        """Return values of original object at the new up-sampling frequency;
+        essentially a re-index with new times set to NaN.
+        """
+        return self._upsample('asfreq')
+
+    def pad(self):
+        """Forward fill new values at up-sampled frequency.
+        """
+        return self._upsample('pad')
+    ffill = pad
+
+    def backfill(self):
+        """Backward fill new values at up-sampled frequency.
+        """
+        return self._upsample('backfill')
+    bfill = backfill
+
+    def nearest(self):
+        """Take new values from nearest original coordinate to up-sampled
+        frequency coordinates.
+        """
+        return self.interpolate('nearest')
+
+    def interpolate(self, kind='linear'):
+        """Interpolate up-sampled data using the original data
+        as knots.
+
+        Parameters
+        ----------
+        kind : str {'linear', 'nearest', 'zero', 'slinear',
+               'quadratic', 'cubic'}
+            Interpolation method to use; refer to to
+            `scipy.interpolate.interp1d` for full details.
+
```

Comment:
Use a `See also` section to reference `scipy.itnerpolate.interp1d` -- these get linked as part of the docs build.